sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/BeamKafkaTable.java

```diff
@@ -84,7 +98,18 @@ public BeamKafkaTable updateConsumerProperties(Map<String, Object> configUpdates
 
   @Override
   public PCollection<Row> buildIOReader(PBegin begin) {
-    KafkaIO.Read<byte[], byte[]> kafkaRead = null;
+    return begin
+        .apply("read", createKafkaRead().withoutMetadata())
+        .apply("in_format", getPTransformForInput())
+        .setRowSchema(getSchema());
+  }
+
+  public static void setNumberOfRecordsForRate(int numberOfRecordsForRate) {
```

Comment:
This looks like a hack, is it possible to keep the `numberOfRecordsForRate` non-static and protected, and then create a setter only in the test table implementation to override in tests?

Reply:
The reason that I included this was giving the user ability to set this before running the query to change the behavior of the estimator. Since user does not have access to the table when the optimization is happening I made it static. 
I'm wondering if the current approach is fine or it is better to make this variable defaultNumberOfRecords and then create one non-static member to be NumberOfRecords. (We can still keep the setter for the static one so that the user can change the default)

Reply:
The problem is we don't know the concrete user scenarios. For one, this table provider can  also be used in JDBC path where users might not have access to the class to mutate it. The specific approach to make this configurable can depend on the use case (e.g. what if the user want to use a different number for different topics or retrieve it from metadata somewhere?) Until we know that I'd rather just make a non-static default, and only override what we need for testing. This way we're not exposing a new API contract and can change the behavior if needed. If we expose an API, users will start using it, and we will have to support it. Without understanding potential use cases I'd rather not commit to that.